generator/internal/rust/templates/common/message.mustache

```diff
@@ -175,11 +175,22 @@ impl gax::paginator::PageableResponse for {{Name}} {
     }
 
     {{/PageableItem}}
+    {{#NextPageToken}}
     fn next_page_token(&self) -> std::string::String {
-        gax::paginator::extract_token(&self.next_page_token)
+        use std::clone::Clone;
+        {{^Optional}}
+        self.{{Codec.FieldName}}.clone()
+        {{/Optional}}
+        {{#Optional}}
+        match &self.{{Codec.FieldName}} {
+            std::option::Option::Some(v) => v.clone(),
+            std::option::Option::None => std::string::String::new(),
+        }
```

Comment:
```suggestion
        self.{{Codec.FieldName}}.cloned().unwrap_or_default()
```

Reply:
Though that may need a `use std::option::Option;`

Reply:
Done.